test/benchee/formatter_test.exs

```diff
@@ -2,16 +2,44 @@ defmodule Benchee.FormatterTest do
   use ExUnit.Case, async: true
   alias Benchee.{Suite, Formatter, Test.FakeFormatter}
 
-  @suite %Suite{}
-  describe "parallel_output/2" do
+  describe "output/1" do
     test "calls `write/1` with the output of `format/1` on each module" do
-      Formatter.parallel_output(@suite, [FakeFormatter])
+      Formatter.output(%Suite{configuration: %{formatters: [{FakeFormatter, %{}}]}})
 
-      assert_receive {:write, "output of `format/1`"}
+      assert_receive {:write, "output of `format/1` with %{}"}
+    end
+
+    test "works with just modules without option tuple" do
+      Formatter.output(%Suite{configuration: %{formatters: [FakeFormatter]}})
+
+      assert_receive {:write, "output of `format/1` with %{}"}
+    end
+
+    test "options are passed on correctly" do
+      Formatter.output(%Suite{configuration: %{formatters: [{FakeFormatter, %{a: :b}}]}})
+
+      assert_receive {:write, "output of `format/1` with %{a: :b}"}
+    end
+
+    test "mixing functions and formatters works" do
+      suite = %Suite{
+        configuration: %{
+          formatters: [
+            {FakeFormatter, %{}},
+            fn _ -> send(self(), {:fun, "me"}) end
```

Comment:
Should we also test that the suite is being passed as the argument here? So, we could do `fn suite -> send(self(), {:fun, suite}) end` and then `assert_receive {:fun, ^suite}`.

Reply:
good idea will add that